pkg/local_object_storage/shard/control.go

```diff
@@ -26,7 +26,17 @@ func (s *Shard) Open() error {
 			return fmt.Errorf("could not open %T: %w", component, err)
 		}
 	}
-	return nil
+
+	return s.openMetabase()
```

Comment:
It changes the order of opening, can we just check whether `component == s.metaBase` in cycle?
The `SetMode` could be done separately in the end.

Also, same applies to `Init`.